hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/SqlConnector.java

```diff
@@ -23,13 +23,13 @@
 import com.hazelcast.jet.core.Vertex;
 import com.hazelcast.jet.sql.impl.ExpressionUtil;
 import com.hazelcast.jet.sql.impl.JetJoinInfo;
-import com.hazelcast.jet.sql.impl.schema.HazelcastTable;
 import com.hazelcast.spi.impl.NodeEngine;
 import com.hazelcast.sql.impl.expression.Expression;
 import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
 import com.hazelcast.sql.impl.row.JetSqlRow;
 import com.hazelcast.sql.impl.schema.MappingField;
 import com.hazelcast.sql.impl.schema.Table;
+import org.apache.calcite.rex.RexInputRef;
 import org.apache.calcite.rex.RexNode;
```

Comment:
do we want to add more Calcite dependencies here? SqlConnector interface is supposed to be public api someday, isn't it?

Reply:
Yes, we wanted to avoid exposure to calcite. But some connectors don't execute the expressions directly, but convert them to something else their target system can understand. Currently this is only JDBC, and you can unparse `RexNode` into SQL, even using various dialects.

Think of it like this. Yes, we wanted to avoid exposure to Calcite, but we also wanted to avoid exposure to `Expression` implementations. We can either make `Expression` and all its implementations a public API, or we can reuse the public API of Calcite. We chose the latter.

Reply:
The point about a shaded dep is fair. The users will have to use our shaded copy. Our control over it is at least the choice whether we upgrade or not. It can also happen that we'll fork Calcite.

I agree that providing our own mapping might be better, but we'll not be able to benefit from calcite's support of dialects and fixes thereof, we'll have to reimplement everything. For this reason I'm fine with exposing (shaded) calcite object. I'm not sure SqlConnector will ever be public API with strict guarantees, maybe one day we'll provide Jet Core API-level guarantee.